json-jackson-v2/src/main/java/tech/ydb/yoj/repository/db/json/JacksonJsonConverter.java

```diff
@@ -0,0 +1,98 @@
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.common.base.Suppliers;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import tech.ydb.yoj.repository.db.common.JsonConverter;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.function.Supplier;
+
+/**
+ * {@link JsonConverter YOJ JSON Converter} implementation using Jackson as the underlying JSON library.
+ * Use it to support JSON-valued fields ({@link tech.ydb.yoj.databind.schema.Column @Column(flatten=false)} composite
+ * objects and dynamic fields with type of interface/abstract class, e.g. {@link java.util.List}):
+ * <blockquote>
+ * <pre>
+ * CommonConverters.defineJsonConverter(new JacksonJsonConverter());
+ * </pre>
+ * </blockquote>
+ * <p>
+ * To customize the {@link ObjectMapper} being created, create a subclass of {@code JacksonJsonConverter} and override
+ * its {@link #createObjectMapper()} method.
+ */
+public class JacksonJsonConverter implements JsonConverter {
+    private final Supplier<ObjectMapper> mapper = Suppliers.memoize(this::createObjectMapper);
+
+    @Override
+    @SneakyThrows
+    public final String toJson(@NonNull Type type, @Nullable Object o) {
+        return mapper().writerFor(mapper().getTypeFactory().constructType(type)).writeValueAsString(o);
+    }
+
+    @Override
+    @SneakyThrows
+    public final Object fromJson(@NonNull Type type, @NonNull String content) {
+        return mapper().readerFor(mapper().getTypeFactory().constructType(type)).readValue(content);
+    }
+
+    @Override
+    @SneakyThrows
+    public final Object fromObject(@NonNull Type type, @Nullable Object content) {
+        JavaType jacksonType = mapper().getTypeFactory().constructType(type);
+        return content != null
+                ? mapper().convertValue(content, jacksonType)
+                : (jacksonType.isCollectionLikeType() ? List.of() : Map.of());
+    }
+
+    public String toString() {
+        return getClass().getSimpleName();
+    }
+    
+    protected final ObjectMapper mapper() {
+        return mapper.get();
+    }
+
+    protected ObjectMapper createObjectMapper() {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setTimeZone(TimeZone.getDefault());
+        mapper.registerModule(new Jdk8Module());
+        mapper.registerModule(new JavaTimeModule());
+        mapper.registerModule(new SimpleModule()
+                .addAbstractTypeMapping(Set.class, LinkedHashSet.class)
+                .addAbstractTypeMapping(Map.class, LinkedHashMap.class)
+                .addAbstractTypeMapping(List.class, ArrayList.class)
+        );
+        mapper.setSerializationInclusion(Include.NON_NULL);
```

Comment:
question: Is a test needed here? I can't decide myself :)

Reply:
It's used in YDB repository tests, so probably not absolutely necessary :-)